libraries/standard/http/utest/http_utest.c

```diff
@@ -78,7 +78,39 @@ typedef struct _headers
 #define HTTP_TEST_INITIALIZED_HEADER_BUFFER_LEN \
     ( HTTP_TEST_MAX_INITIALIZED_HEADER_LEN + 1 )
 
-#define HTTP_TEST_BUFFER_LEN    1024
+/* Template HTTP header fields and values. */
+#define HTTP_TEST_HEADER_FIELD               "Authorization"
+#define HTTP_TEST_HEADER_FIELD_LEN           ( sizeof( HTTP_TEST_HEADER_FIELD ) - 1 )
+#define HTTP_TEST_HEADER_VALUE               "None"
+#define HTTP_TEST_HEADER_VALUE_LEN           ( sizeof( HTTP_TEST_HEADER_VALUE ) - 1 )
+/* Template for first line of HTTP header. */
+#define HTTP_TEST_HEADER_REQUEST_LINE        "GET / HTTP/1.1 \r\n"
+#define HTTP_TEST_HEADER_REQUEST_LINE_LEN    ( sizeof( HTTP_TEST_HEADER_REQUEST_LINE ) - 1 )
+#define HTTP_REQUEST_HEADERS_INITIALIZER     { 0 }
+/* Template for snprintf(...) strings. */
+#define HTTP_TEST_SINGLE_HEADER_FORMAT       "%s%s: %s\r\n\r\n"
+#define HTTP_TEST_DOUBLE_HEADER_FORMAT       "%s%s: %s\r\n%s: %s\r\n\r\n"
+
+/* Length of the following template HTTP header.
+ *   <HTTP_TEST_HEADER_REQUEST_LINE> \r\n
+ *   <HTTP_TEST_HEADER_FIELD>: <HTTP_TEST_HEADER_VALUE> \r\n
+ *   \r\n
+ * This is used to initialize the expectedHeader string. */
+#define HTTP_TEST_SINGLE_HEADER_LEN       \
+    ( HTTP_TEST_HEADER_REQUEST_LINE_LEN + \
+      HTTP_TEST_HEADER_FIELD_LEN +        \
+      HTTP_HEADER_FIELD_SEPARATOR_LEN +   \
+      HTTP_TEST_HEADER_VALUE_LEN +        \
+      HTTP_HEADER_LINE_SEPARATOR_LEN +    \
+      HTTP_HEADER_LINE_SEPARATOR_LEN )
+
+/* The longest possible header used for these unit tests. */
+#define HTTP_TEST_DOUBLE_HEADER_LEN     \
+    ( HTTP_TEST_SINGLE_HEADER_LEN +     \
+      HTTP_TEST_HEADER_FIELD_LEN +      \
+      HTTP_HEADER_FIELD_SEPARATOR_LEN + \
+      HTTP_TEST_HEADER_VALUE_LEN +      \
+      HTTP_HEADER_LINE_SEPARATOR_LEN )
 
 /* File-scoped Global variables */
 static HTTPStatus_t retCode = HTTP_NOT_SUPPORTED;
```

Comment:
We spoke about removing this MACRO

Reply:
Oh that's only being used for Archit's unit tests, but I'll change it.